### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by rex4539**

### CODE REVIEW

File: docs/SequencesAndCollections.rst [hunk 1]
```diff
@@ -285,7 +285,7 @@ stable indices, the element order within the collection itself is stable; the
 order sometimes does not have a meaning and is not chosen by the code that uses
 the collection, but by the implementation details of the collection itself.)
 
-`MutableCollectionType` protocol allows the to replace a specific element,
+`MutableCollectionType` protocol allows to replace a specific element,
```
Comment (compnerd) [inline docs/SequencesAndCollections.rst]: This still doesn't make sense .. perhaps it should be `allows the caller`?

File: docs/ARCOptimization.rst [hunk 2]
```diff
@@ -544,7 +544,7 @@ optimization? We must consider three areas of concern:
    retain/release counts in the loop? Consider a set of retains and a set of
    releases that we wish to hoist out of a loop. We can only hoist the retain,
    release sets out of the loop if all paths in the given loop region from the
-   entrance to the backedge.  have exactly one retain or release from this set.
+   entrance to the backedge. Have exactly one retain or release from this set.
```
Comment (compnerd) [inline docs/ARCOptimization.rst]: This is still a fragment :-(.

Comment (xwu) [reply docs/ARCOptimization.rst]:
```suggestion
   entrance to the backedge have exactly one retain or release from this set.
```

File: docs/LogicalObjects.rst [hunk 3]
```diff
@@ -40,7 +40,7 @@ otherwise they're pretty much just like they are in Objective-C.
 That said, they do interact with some other language features in interesting
 ways.
 
-For example, methods on value types have a this parameter.  Usually parameters
```
Comment (xwu) [inline docs/LogicalObjects.rst]: This is correct as-is. It could use code voice for clarity but it is not necessary to be interpretable.

**Comment by rex4539** [comment]
All comments are addressed.

**Comment by CodaFi** [comment]
LGTM.  Thanks!

⛵️ 

@swift-ci please smoke test and merge

**Comment by CodaFi** [comment]
Corelibs-Foundation test runner segfaulted.  I'll try re-running those tests manually later.

**Comment by CodaFi** [comment]
@swift-ci please smoke test Linux platform

**Comment by rex4539** [comment]
Anything else required from my side or is this ready for merging?

**Comment by CodaFi** [comment]
Nope, all good.

⛵️

**Pull request merged**